stan/math/prim/mat/fun/ocl_gpu.hpp

```diff
@@ -0,0 +1,296 @@
+ * hold the OpenCL kernel sources, the groups to
+ * which each kernel is assigned to and the
+ * information about which kernel was already compiled.
+ *
+ */
+inline void init_kernel_groups() {
+  // To identify the kernel group
+  kernel_groups["transpose"] = "basic_matrix";
+  kernel_groups["copy"] = "basic_matrix";
+  kernel_groups["zeros"] = "basic_matrix";
+  kernel_groups["identity"] = "basic_matrix";
+  kernel_groups["copy_triangular"] = "basic_matrix";
+  kernel_groups["copy_triangular_transposed"] = "basic_matrix";
+  kernel_groups["add"] = "basic_matrix";
+  kernel_groups["subtract"] = "basic_matrix";
+  kernel_groups["copy_submatrix"] = "basic_matrix";
+  kernel_groups["scalar_mul_diagonal"] = "matrix_multiply";
+  kernel_groups["scalar_mul"] = "matrix_multiply";
+  kernel_groups["basic_multiply"] = "matrix_multiply";
+  kernel_groups["multiply_self_transposed"] = "matrix_multiply";
+  kernel_groups["lower_tri_inv_step1"] = "matrix_inverse";
+  kernel_groups["lower_tri_inv_step2"] = "matrix_inverse";
+  kernel_groups["lower_tri_inv_step3"] = "matrix_inverse";
+  kernel_groups["cholesky_block"] = "cholesky_decomposition";
+  kernel_groups["check_nan"] = "check_gpu";
+  kernel_groups["check_symmetric"] = "check_gpu";
+  kernel_groups["check_diagonal_zeros"] = "check_gpu";
+
+  kernel_groups["dummy"] = "timing";
+
+  // Kernel group strings
+  // the dummy kernel is the only one not included in files
+  // so it is treated before the loop that iterates
+  // through  kernels to load all
+
+  kernel_strings["timing"] = dummy_kernel;
+  kernel_strings["check_gpu"] =
+#include <stan/math/prim/mat/fun/kern_gpu/check_gpu.cl>  // NOLINT
+      ;                                                  // NOLINT
+  kernel_strings["cholesky_decomposition"] =
+#include <stan/math/prim/mat/fun/kern_gpu/cholesky_decomposition.cl>  // NOLINT
+      ;                                                               // NOLINT
+  kernel_strings["matrix_inverse"] =
+#include <stan/math/prim/mat/fun/kern_gpu/matrix_inverse.cl>  // NOLINT
+      ;                                                       // NOLINT
+  kernel_strings["matrix_multiply"] =
+#include <stan/math/prim/mat/fun/kern_gpu/matrix_multiply.cl>  // NOLINT
+      ;                                                        // NOLINT
+  kernel_strings["basic_matrix"] =
+#include <stan/math/prim/mat/fun/kern_gpu/basic_matrix.cl>  // NOLINT
+      ;                                                     // NOLINT
+
+  // Check if the kernels were already compiled
+  compiled_kernels["basic_matrix"] = false;
+  compiled_kernels["matrix_multiply"] = false;
+  compiled_kernels["timing"] = false;
+  compiled_kernels["matrix_inverse"] = false;
+  compiled_kernels["cholesky_decomposition"] = false;
+  compiled_kernels["check_gpu"] = false;
+}
+
+// TODO(Rok): select some other platform/device than 0
+// TODO(Rok): option to turn profiling OFF
+/**
+ * The class that represents the OpenCL runtime
```

Comment:
Update doc. There's a lot that's in this PR that should be inside this documentation.